xchange-coingi/src/main/java/org/knowm/xchange/coingi/dto/account/CoingiBalance.java

```diff
@@ -0,0 +1,90 @@
+package org.knowm.xchange.coingi.dto.account;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import java.math.BigDecimal;
+import java.util.Map;
+import java.util.Objects;
+
+public class CoingiBalance {
+  private Map<String, Object> currency;
+
+  private BigDecimal available;
+
+  private BigDecimal inOrders;
+
+  private BigDecimal deposited;
+
+  private BigDecimal withdrawing;
+
+  private BigDecimal blocked;
+
+  public CoingiBalance(
+      @JsonProperty("currency") Map<String, Object> currency,
+      @JsonProperty("available") BigDecimal available,
+      @JsonProperty("inOrders") BigDecimal inOrders,
+      @JsonProperty("deposited") BigDecimal deposited,
+      @JsonProperty("withdrawing") BigDecimal withdrawing,
+      @JsonProperty("blocked") BigDecimal blocked) {
+
+    this.currency = currency;
+    this.available = available;
+    this.inOrders = inOrders;
+    this.deposited = deposited;
+    this.withdrawing = withdrawing;
+    this.blocked = blocked;
+  }
+
+  CoingiBalance() {}
+
+  public Map<String, Object> getCurrency() {
+    return currency;
+  }
+
+  public String getCurrencyName() {
+    return (String) currency.get("name");
+  }
+
+  public boolean getCurrencyCrypto() {
+    return (Boolean) currency.get("crypto");
+  }
```

Comment:
lets add a `CoingiBalanceCurrency` DTO instead of using a Map

Reply:
done, thanks

Reply:
done